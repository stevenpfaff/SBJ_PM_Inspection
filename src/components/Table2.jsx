import { Table } from 'react-bootstrap'

import tasks from '../data/table2.json'

const Table2 = () => (
  <Table striped bordered hover>
    <thead>
      <tr>
        <th>Task</th>
        <th>Completed</th>
        <th>Not Performed</th>
        <th>Part #</th>
      </tr>
    </thead>
    <tbody>
      {tasks.map((task, idx) => (
        <tr key={idx}>
          <td>{task.name}</td>
          <td>
            <input
              type='checkbox'
              checked={task.completed}
              onChange={() => {}}
              data-testid='table-2-completed'
            />
          </td>
          <td>
            <input
              type='checkbox'
              checked={task.notPerformed}
              onChange={() => {}}
              data-testid='table-2-not-performed'
            />
          </td>
          <td className='fixed-width'>{task.partNumber}</td>
        </tr>
      ))}
    </tbody>
  </Table>
)

export default Table2
