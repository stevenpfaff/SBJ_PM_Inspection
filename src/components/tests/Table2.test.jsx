import { fireEvent, render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

import Table2 from '../Table2'

describe('Table2', () => {
  it('should match snapshot', () => {
    const { container } = render(<Table2 />)

    expect(container).toMatchSnapshot()
  })

  it('should do nothing on "completed" checkbox click', () => {
    const { asFragment, getAllByTestId } = render(<Table2 />)

    const completed = getAllByTestId('table-2-completed')

    expect(completed.length).not.toBeFalsy()

    const old = asFragment()

    act(() => {
      fireEvent.click(completed[0])
    })

    expect(asFragment()).toStrictEqual(old)
  })

  it('should do nothing on "not performed" checkbox click', () => {
    const { asFragment, getAllByTestId } = render(<Table2 />)

    const notPerformed = getAllByTestId('table-2-not-performed')

    expect(notPerformed.length).not.toBeFalsy()

    const old = asFragment()

    act(() => {
      fireEvent.click(notPerformed[0])
    })

    expect(asFragment()).toStrictEqual(old)
  })
})
