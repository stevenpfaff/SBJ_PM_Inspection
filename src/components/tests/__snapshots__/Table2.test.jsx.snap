// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table2 should match snapshot 1`] = `
<div>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Task
        </th>
        <th>
          Completed
        </th>
        <th>
          Not Performed
        </th>
        <th>
          Part #
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Oil Change
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          A320HGOEJ4
        </td>
      </tr>
      <tr>
        <td>
          Fuel Filters
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          6OF8WBFY6K
        </td>
      </tr>
      <tr>
        <td>
          Grease
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          M7R0S8PTRR
        </td>
      </tr>
      <tr>
        <td>
          Air Filter
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          HRXILKADTP
        </td>
      </tr>
      <tr>
        <td>
          Trans Service / Filter
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          GA9UPACDT9
        </td>
      </tr>
      <tr>
        <td>
          Axle Service
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          NUAI6HW36C
        </td>
      </tr>
      <tr>
        <td>
          Hydraulic Service / Filter
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          NO319JFOHF
        </td>
      </tr>
      <tr>
        <td>
          Drain Air Tanks
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          W221SNJFVX
        </td>
      </tr>
      <tr>
        <td>
          Apply PM Sticker
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          67V86C503B
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
